server/games/typedefs.py

```diff
@@ -154,13 +155,25 @@ def from_basic(
         validity: ValidityState,
         team_outcomes: List[GameOutcome],
         commander_kills: Dict[str, int],
+        team_partial_army_results: List[List[ArmyResult]],
     ) -> "EndedGameInfo":
         if len(basic_info.teams) != len(team_outcomes):
             raise ValueError(
                 "Team sets of basic_info and team outcomes must refer to the "
                 "same number of teams in the same order."
             )
 
+        # team_partial_army_results were initialized with ArmyOutcome.UNKNOWN so
+        # we need to get the resolved outcome from team_outcomes
+        team_complete_army_results = []
+        for outcome, army_results in zip(team_outcomes, team_partial_army_results):
+            complete_team_results = []
+            for player in army_results:
+                resolved_data = {**player._asdict(), "army_result": outcome.name}
```

Comment:
Ahh, I completely misunderstood that, thank you for catching it!

Reply:
The result reporting/resolution is a bit confusing since the words victory/defeat/draw have different meanings in different contexts. That was the main reason for creating 3 different types for them so it would be easier to keep track of which context you're dealing with.